packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,294 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { IContainer } from "@fluidframework/container-definitions";
+import {
+    ContainerRuntime,
+    IContainerRuntimeOptions,
+    IGCRuntimeOptions,
+    ISummarizer,
+} from "@fluidframework/container-runtime";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+    ITestObjectProvider,
+    createSummarizerWithContainer,
+    summarizeNow,
+    waitForContainerConnection,
+    mockConfigProvider,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, itExpects } from "@fluidframework/test-version-utils";
+import { DataObject, DataObjectFactory, ContainerRuntimeFactoryWithDefaultDataStore } from "@fluidframework/aqueduct";
+import { delay } from "@fluidframework/common-utils";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { IContainerRuntimeBase } from "@fluidframework/runtime-definitions";
+
+class TestDataObject extends DataObject {
```

Comment:
Don't create a separate data object unless you need functionality that is missing by the data object provided by the test object provider.
The problem with this is that you will lose back-compat testing. See https://github.com/microsoft/FluidFramework/pull/12052. You should directly call `provider.makeTestContainer` or `provider.loadTestContainer` and use the compat version of the test data object it provides.

Reply:
Created a task to do this, https://dev.azure.com/fluidframework/internal/_workitems/edit/2320, the only reason why I created my own testDataObject and factory was so that I could create datastores via the factory method.

Reply:
Via offline conversation, this was fixed